### ISSUE TITLE
Android: Support AppManager shortcuts

### ISSUE THREAD

**Issue by shuvashish76** [samuelclay/NewsBlur#1624]
Shortcut created using [AppManager](https://github.com/MuntashirAkon/AppManager)
App crashes when I click on the shortcut from the launcher.

![Screenshot_2022-02-10-00-21-11](https://user-images.githubusercontent.com/48226312/153269805-7597323c-e272-42ba-a590-da31ad11d684.jpeg)

![Screenshot_2022-02-10-00-21-28](https://user-images.githubusercontent.com/48226312/153269901-46f2ced8-6a39-4662-b9bb-1a11a2739c6d.jpeg)

Crash logs:

```
FATAL EXCEPTION: main
Process: com.newsblur, PID: 18636
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.newsblur/com.newsblur.activity.AddFeedExternal}: java.lang.NullPointerException
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2781)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2859)
	at android.app.ActivityThread.-wrap11(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1592)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6518)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
Caused by: java.lang.NullPointerException
	at com.newsblur.util.FeedUtils.dropAndRecreateTables(FeedUtils.kt:72)
	at com.newsblur.util.PrefsUtils.logout(PrefsUtils.java:182)
	at com.newsblur.activity.NbActivity.onCreate(NbActivity.kt:43)
	at com.newsblur.activity.AddFeedExternal.onCreate(AddFeedExternal.java:23)
	at android.app.Activity.performCreate(Activity.java:7023)
	at android.app.Activity.performCreate(Activity.java:7014)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2734)
	... 9 more
```

App version: `11.2.2 (201)`

**Comment by sictiru** [comment]
@shuvashish76 The crash itself was fixed but it's in a part of the app that I'm not sure it's used anymore. In the stacktrace you can see that it performs a logout which would close the app in this case.

Why does the shortcut open `AddFeedExternal`? What was your initial goal with the shortcut created with that tool?
In case you are looking to just create a launch shortcut, I'd check if the tool picks up `InitActivity` as the entry point and use that.

**Comment by shuvashish76** [comment]
@samuelclay 
> do we support that? We have similar on iOS:

Oh wow! No we do not support shortcuts. Currently we do have 1 widget though.

![Screenshot_2022-02-17-00-19-33](https://user-images.githubusercontent.com/48226312/154337966-aa370e78-33b6-4dcf-806e-bbb3dddb803a.jpeg)
![Screenshot_2022-02-17-00-09-30](https://user-images.githubusercontent.com/48226312/154337986-06b8c731-5398-4e4e-94a9-dbf22df61674.jpeg)

I've seen other apps use both shortcuts & widgets.

![Screenshot_2022-02-17-00-21-57](https://user-images.githubusercontent.com/48226312/154338144-d151a67f-47cd-40cc-9773-2af3a8ed352e.jpeg)
![Screenshot_2022-02-17-00-22-12](https://user-images.githubusercontent.com/48226312/154338170-e512a629-5832-448d-87e1-76bee3cbfc62.jpeg)
screenshorts source: [Bromite](https://github.com/bromite/bromite)

I think shortcuts would be a nice addition to android version.

---

@sictiru 
> What was your initial goal with the shortcut created with that tool?

I was looking for hidden shortcuts _(as what samuelclay mentioned in the previous comment)_

> In case you are looking to just create a launch shortcut, I'd check if the tool picks up InitActivity as the entry point and use that.

Nope launch shortcut isn't useful for me.
Yes AppManager show that `InitActivity` as well :)

![Screenshot_2022-02-17-00-36-21](https://user-images.githubusercontent.com/48226312/154339083-65aaed0f-c090-4b12-b171-64b2e0bb1544.jpeg)

Edit Shortcut 👇🏿

![Screenshot_2022-02-17-00-36-52](https://user-images.githubusercontent.com/48226312/154339120-d3a09c98-3c37-41fc-8770-6d62b291b201.jpeg)

**Comment by sictiru** [comment]
Shortcuts similar to the iOS app can be added to Android too but probably not for all of them due to inter app dependencies.

**Comment by sictiru** [comment]
Indeed it is crashing since the intent filters are not matched when opening with AM. 
Check added for empty URI and it will redirect to main page if necessary.

**Comment by sictiru** [comment]
It looks like the shortcut icons are rendered differently between launchers. I will add SVGs which should help with the icon formatting.

**Comment by sictiru** [comment]
@shuvashish76 Out of 5 launchers, just Lawnchair Launcher didn't render the icons properly for some reason. Also it didn't like just any SVG ¯\_(ツ)_/¯ so new icons are used for shortcuts.

**Comment by samuelclay** [comment]
New icons are available under `/media/img/icons/nouns` on the `pro` branch. I should be launching it within the next couple days, so it'll be merged into master soon enough.

**Comment by sictiru** [comment]
https://github.com/samuelclay/NewsBlur/issues/1694 was released in 12.0 but Lawnchair Launcher didn't like the new icons either so I added another search icon just for the shortcut to make it work consistently.

**Comment by sictiru** [comment]
Latest changes are live in 12.0.1
